### PR TITLE
Lift the first child into an option to prevent 500 if none exist

### PR DESCRIPTION
**Pull request by gtrufitt**
Easier to see changes with https://github.com/guardian/frontend/pull/15506/files?w=0

## What does this change?

Probably not anything as it's not possible to cause this error anymore in composer *but* fixes it historic screw ups I guess:

"Unlikely there is an existing example of this any more, as we should have fixed the one that was broken. Essentially we need to make the child array access safe in L98-101 of VideoEmbedCleaner - just in case a video is embedded in such a way that there is no child element of an "element-video"."

## What is the value of this and can you measure success?

Bug fix, moves a Trello card to Done.

### CODE REVIEW

File: common/app/views/support/cleaner/VideoEmbedCleaner.scala [hunk 1]
```diff
@@ -96,16 +96,17 @@ case class VideoEmbedCleaner(article: Article) extends HtmlCleaner {
       element.getElementsByClass("gu-video").isEmpty
     }.foreach { element: Element =>
       val canonicalUrl = element.attr("data-canonical-url")
+      element.children().lift(0).map(child => {
```
Comment (jfsoul) [inline common/app/views/support/cleaner/VideoEmbedCleaner.scala]: Nice use of `lift` 

Would marginally prefer `.lift(0).map { ... }` notation for consistency though

**Review by jfsoul — APPROVED** [review]
Looks good, minor comment aside 👍

File: common/app/views/support/cleaner/VideoEmbedCleaner.scala (hunk 1 above)
Comment (TBonnin) [inline common/app/views/support/cleaner/VideoEmbedCleaner.scala]: would prefer `headOption` but not a big deal

Comment (TBonnin) [inline common/app/views/support/cleaner/VideoEmbedCleaner.scala]: Actually `collectFirst` would be better :D

**Review by TBonnin — CHANGES_REQUESTED** [review]
See my comment about `collectFirst`

**Comment by gtrufitt** [comment]
@TBonnin Replaced with `headOption.foreach` as per our non-PR chat as this function has side effects https://github.com/guardian/frontend/pull/15506/files?w=1

File: common/app/views/support/cleaner/VideoEmbedCleaner.scala [hunk 2]
```diff
@@ -106,7 +106,7 @@ case class VideoEmbedCleaner(article: Article) extends HtmlCleaner {
         }
 
         child.wrap("<div class=\"embed-video-wrapper u-responsive-ratio u-responsive-ratio--hd\"></div>")
-      });
+      };
```
Comment (LATaylor-guardian) [inline common/app/views/support/cleaner/VideoEmbedCleaner.scala]: Why is there a semi colon here?

Comment (gtrufitt) [inline common/app/views/support/cleaner/VideoEmbedCleaner.scala]: Because I'm a client-side developer :man_facepalming:

**Review by TBonnin — CHANGES_REQUESTED** [review]
👍

**Review by TBonnin — APPROVED** [review]
👍

**Pull request merged**